src/internal_modules/roc_fec/reader.cpp

```diff
@@ -789,5 +804,24 @@ void Reader::drop_repair_packets_from_prev_blocks_() {
     }
 }
 
+void Reader::update_block_duration_(const packet::PacketPtr& ptr) {
+    if (!ptr->rtp()) {
+        return;
+    }
```

Comment:
Let's use `Packet::stream_timestamp()` instead of `Packet::rtp()`, this way fec::Reader will continue to be RTP-independent.